spirv_msl.cpp

```diff
@@ -67,6 +67,32 @@ void CompilerMSL::add_msl_resource_binding(const MSLResourceBinding &binding)
 {
 	StageSetBinding tuple = { binding.stage, binding.desc_set, binding.binding };
 	resource_bindings[tuple] = { binding, false };
+
+	// If we might need to pad argument buffer members to positionally align
+	// arg buffer indexes, also maintain a lookup by argument buffer index.
+	if (msl_options.pad_argument_buffer_resources)
+	{
+		uint32_t arg_buff_idx = k_unknown_component;
+		switch (binding.base_type)
+		{
+		case SPIRType::Void:
+			arg_buff_idx = binding.msl_buffer;
+			break;
+		case SPIRType::Image:
+		case SPIRType::SampledImage:
+			arg_buff_idx = binding.msl_texture;
```

Comment:
I think we need to split SampledImage into binding.msl_texture and binding.msl_sampler if I recall the implementation correctly.

Reply:
I don't think that's needed. However...in thinking more about your question above about distribution of ID's between the texture and sampler component...we might be able to. Let me think about this and see if a simple option exists.

Reply:
Latest adds separate lookups for the image and sampler components of `SampledImage`.